File: packages/services/pkg/protocol-parser/staticDataLength.go
package protocolparser

import schematype "latticexyz/mud/packages/services/pkg/schema-type"

func StaticDataLength(staticFields []schematype.SchemaType) int {
	var length int
	for _, field := range staticFields {
		length += schematype.StaticAbiTypeToByteLength(field)
	}
	return length
}

File: packages/protocol-parser/src/staticDataLength.ts
import { StaticAbiType, staticAbiTypeToByteLength } from "@latticexyz/schema-type";

export function staticDataLength(staticFields: readonly StaticAbiType[]): number {
  return staticFields.reduce((length, fieldType) => length + staticAbiTypeToByteLength[fieldType], 0);
}
